pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,491 @@
+        "B",
+        """Return selected connectors ("front"/"back"), based on front-panel selector switch
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "front_rear")),
+        values={"back":0, "front":1},
+        map_values = True,
+        )
+
+    auto_range = Instrument.measurement(
+        "B",
+        """ Return auto-ranging status, returns False if manual range and True if auto-range active.
+        For manual range control the range property can be set
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "auto_range")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    auto_zero = Instrument.control(
+        "B",
+        "Z%d",
+        """ Returns autozero settings on the HP3478, this property can be set 
+        (False: disabled, True: enabled) 
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "auto_zero")),
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    cal_enable = Instrument.measurement(
+        "B",
+        """Return calibration enable switch setting (False: cal disabled, True: cablibration possible),
+        based on front-panel selector switch
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "cal_enable")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        :return error_status: one byte with the error status register content
+        :rtype error_status: int
+        """
+
+        if self.error_status != 0:
+            log.critical("HP3478A error detected: $s", self.ERRORS[self.error_status] )
+        return self.error_status
+
+    error_status = Instrument.measurement(
+        "E",
+        """Checks the error status register
+        """,
+        get_process = (lambda x: int(x)),
+        )
+
+    disp_reset = Instrument.setting(
+        "D%d",
+        """Resets the display of the instrument if 1 is written to it.
+        """,
+        set_process = (lambda x: int(x)),
+        validator = strict_discrete_set,
+        values = [1],
+        )
+
+    #TODO: find a proper validator for the next two methods (without using regex, if possible)
+    disp_text = Instrument.setting(
+        "D2%s",
+        """Displays up to 12 upper-case ASCII characters on the display.
+        """,
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    disp_text_no_symbol = Instrument.setting(
+        "D3%s",
+        """Displays up to 12 upper-case ASCII characters on the display and disables all symbols on the display.
+        """,
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    #TODO: find a good way to cache the latest measurment and use it to incease speed (if actually needed)
```

Comment:
I'm inclined to think that caching the measurement result is not something that should be done within the instrument implementation, because in the end that is up to the user to decide whether and when a cached result should be used, or do you disagree?

Reply:
Yes, 
I remove this TODO